Plugins/NativeEngine/Source/ShaderCache.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <Babylon/ShaderCache.h>
+
+#include <iostream>
+#include <fstream>
+
+namespace Babylon
+{
+    class ShaderCache::Impl
+    {
+    public:
+        Impl() = default;
+        ~Impl() = default;
+        const ShaderCompiler::BgfxShaderInfo* GetShader(std::string_view vertexSource, std::string_view fragmentSource);
+        void AddShader(std::string_view vertexSource, std::string_view fragmentSource, ShaderCompiler::BgfxShaderInfo);
+
+        void SerializeOutTo(std::ofstream& stream);
+        uint32_t SerializeInFrom(std::ifstream& stream);
```

Comment:
I think `Serialize` and `Deserialize` would be more standard naming?

Reply:
Makes sense.fixed